src/seedu/addressbook/data/person/Address.java

```diff
@@ -27,6 +32,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         }
         this.value = address;
     }
+    
+    public Address(int blockNumber, String streetName, String unitNumber, int postalNumber) {
```

Comment:
As discussed in class, its better to split the address within the constructor. Also, you have missed updating calls to the constructor in other files. Try regression testing to discover further changes you need to make.